ref-farming/src/actions_of_reward.rs

```diff
@@ -168,21 +167,17 @@ impl Contract {
         let view_farmer = self.get_farmer(&farmer_id);
         let tokens = {
             if let Some(token) = token {
-                vec![token]
+                vec![token.clone()]
             } else {
-                view_farmer.get_ref().rewards.keys().collect::<Vec<_>>()
+                view_farmer.get_ref().rewards.keys().map(|x| x.clone()).collect::<Vec<_>>()
             }
         };
```

Comment:
Why clone it, if you still borrowing it below? Is it because of the immutable/mutable burrow later?

Reply:
Yes, I use it mutable later. It's like to get an independent snapshot of all reward tokens into `tokens`.